### PR TITLE
feat: new error types for destinations (webhook, wootric, woopra, yahoo_dsp, zendesk)

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/webhook/transform.js [hunk 1]
```diff
@@ -154,11 +157,10 @@ function process(event) {
...
       return response;
     }
-    throw new CustomError("Invalid URL in destination config", 400);
+    throw new ConfigurationError("Invalid URL in destination config");
   } catch (err) {
-    throw new CustomError(
-      err.message || "[webhook] Failed to process request",
-      err.status || 400
+    throw new NetworkInstrumentationError(
+      err.message || "[webhook] Failed to process request"
```
Comment (koladilip) [inline src/v0/destinations/webhook/transform.js:163]: remove [webhook] prefix

**Review by koladilip — APPROVED** [review]

File: src/v0/destinations/webhook/transform.js (hunk 1 above)
Comment (saikumarrs) [inline src/v0/destinations/webhook/transform.js]: Remove the outer-level try-catch in this method.

File: src/v0/destinations/wootric/util.js [hunk 2]
```diff
@@ -55,11 +58,10 @@ const getAccessToken = async destination => {
     const processedAuthResponse = processAxiosResponse(wootricAuthResponse);
     // If the request fails, throwing error.
     if (processedAuthResponse.status !== 200) {
-      throw new CustomError(
-        `[Wootric]:: access token could not be generated due to ${JSON.stringify(
+      throw new NetworkInstrumentationError(
```
Comment (saikumarrs) [inline src/v0/destinations/wootric/util.js:61]: Is this clearly an instrumentation error?

File: src/v0/destinations/wootric/transform.js [hunk 3]
```diff
@@ -45,7 +51,7 @@ const identifyResponseBuilder = async (message, destination) => {
 
   const accessToken = await getAccessToken(destination);
   if (!accessToken) {
-    throw new CustomError("Access token is missing", 400);
+    throw new NetworkInstrumentationError("Access token is missing");
   }
```
Comment (saikumarrs) [inline src/v0/destinations/wootric/transform.js:55]: Based on the `getAccessToken` implementation, this validation may not be required. Double-check that, please.
Also, even if we were to throw any error here, it is definitely not the instrumentation error.

File: src/v0/destinations/wootric/transform.js [hunk 4]
```diff
@@ -85,7 +91,7 @@ const trackResponseBuilder = async (message, destination) => {
 
   const accessToken = await getAccessToken(destination);
   if (!accessToken) {
-    throw new CustomError("Access token is missing", 400);
+    throw new NetworkInstrumentationError("Access token is missing");
```
Comment (saikumarrs) [inline src/v0/destinations/wootric/transform.js:94]: Same as above.

File: src/v0/destinations/wootric/util.js [hunk 5]
```diff
@@ -105,11 +106,10 @@ const retrieveUserDetails = async (endUserId, externalId, accessToken) => {
   }
 
   if (processedUserResponse.status !== 404) {
-    throw new CustomError(
-      `[Wootric]:: Unable to retrieve userId due to ${JSON.stringify(
+    throw new NetworkInstrumentationError(
```
Comment (saikumarrs) [inline src/v0/destinations/wootric/util.js:109]: Is this really an instrumentation error? Also, the status is ignored here.

File: src/v0/destinations/webhook/transform.js (hunk 1 above)
Comment (mihir-4116) [reply src/v0/destinations/webhook/transform.js]: updated

File: src/v0/destinations/yahoo_dsp/transform.js [hunk 6]
```diff
@@ -81,7 +81,7 @@ const responseBuilder = async (message, destination) => {
   response.method = defaultPutRequestConfig.requestMethod;
   const accessToken = await getAccessToken(destination);
   if (!accessToken) {
-    throw new CustomError(`[Yahoo_DSP]:: access token is not available`, 400);
+    throw new NetworkInstrumentationError("Access token is not available");
```
Comment (saikumarrs) [inline src/v0/destinations/yahoo_dsp/transform.js:84]: Same as above.

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: src/v0/destinations/yahoo_dsp/transform.js (hunk 6 above)
Comment (saikumarrs) [reply src/v0/destinations/yahoo_dsp/transform.js:84]: Is this addressed?

Comment (mihir-4116) [reply src/v0/destinations/yahoo_dsp/transform.js]: updated

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**